crates/utils/src/network_note.rs

```diff
@@ -0,0 +1,49 @@
+use miden_objects::note::{Note, NoteExecutionMode, NoteId, NoteMetadata, NoteTag, Nullifier};
+use thiserror::Error;
+
+/// A newtype that wraps around notes targeting a single account to be used in a network mode.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct NetworkNote(Note);
```

Comment:
Wasn't too sure where to place this since it could eventually end up being used in various different crates, and this seemed like the less problematic one, but I could move it to the NTB crate if preferred